h/views/api_auth.py

```diff
@@ -85,6 +112,22 @@ def _authorized_response(self):
             client_id = self.request.params.get('client_id')
             raise RuntimeError('created authorisation code for client "{}" but got no redirect location'.format(client_id))
 
+    def _render_web_message_response(self, redirect_uri):
```

Comment:
Same as above, `oauthlib` doesn't support `web_message` and its API is not designed to make this easy, the only way to get to the authorization code is to parse the redirect URI.